src/main/java/com/scaler/lld/design/assignments/builder/DatabaseConfigurationBuilder.java

```diff
@@ -2,5 +2,66 @@
 
 @WithBuilder
 public class DatabaseConfigurationBuilder {
+    private String databaseUrl;
+    private String username;
+    private String password;
+    private int maxConnections;
+    private boolean enableCache;
+    private boolean isReadOnly;
+
+    //step 1:hide the constructor
+    private DatabaseConfigurationBuilder(){
+
+    }
+
+    public static DBConfigBuilder BuildDBConfiguration(){
+        return new DBConfigBuilder();
+    }
+
+    public static class DBConfigBuilder{
+        private DatabaseConfigurationBuilder instance;
+
+        public DBConfigBuilder( ){
+
+            this.instance = new DatabaseConfigurationBuilder();
+        }
+
+        public DBConfigBuilder WithDatabaseUrl(String url){
```

Comment:
[Minor] Java uses camel case so a method name should look like `withDatabaseUrl`.
Read more about it [here](https://www.theserverside.com/answer/Pascal-case-vs-camel-case-Whats-the-difference).